### DISCUSSION TITLE
Program using Magick++ library freezes when reading file from disk

### DISCUSSION THREAD

**Discussion by danielzx2** [ImageMagick/ImageMagick#7164] [Help] [answered]
### ImageMagick version

7.1.1-Q16

### Operating system, version and so on

Windows 10 22H2

### Description

I am currently trying to use Magick++ for a personal project. I have the library set up that all DLLs and lib files are configured and ready for use. Then I tried experimenting with it by writing a program that

1. Constructs an Image object
2. Reads an image file (tried both jpg and png)
3. Performs crop operation
4. Writes to a new png file named result.png

When running the program through the debugger, it freezes and seems to go into an endless loop when trying to read a file from disk. Is there something I'm missing?
My code looks like this, for reference:

```
int main(int argc, char** argv)
{
    try
    {
        Magick::InitializeMagick(*argv);
        Magick::Image test;
        test.read("20220704234052_1.jpg");
        if (test.isValid())
        {
            std::cout << "Image loaded \n";
            test.crop((Magick::Geometry(20, 20, 20, 20)));
            test.write("result.png");
            std::cout << "Operation Complete!";
        }
    }
    catch (Magick::Exception &error_)
    {
        std::cout << "Caught Error: " << error_.what() << std::endl;
        return 1;
    }
    return 0;
}
```

**Comment by snibgo** [comment]
I suggest you zip `20220704234052_1.jpg` and post that here.

**Reply by danielzx2** [reply]
[20220704234052_1.zip](https://github.com/ImageMagick/ImageMagick/files/14563526/20220704234052_1.zip)

This is the image I use.

When testing, I had it in the same folder as the .cpp file

**Comment by urban-warrior** [comment]
We tried your Magick++ program on our system and it completed in less than a tenth of a second.

**Comment by danielzx2** [comment]
Could you attach the resulting image here?

If so, was it some dll och linked which causes it crash in my end? This has happened on two machines. Both were using Microsoft Visual Studio 2022.

**Comment by danielzx2** [comment]
Found more information about the error:

Microsoft C++ exception: Magick::ErrorMissingDelegate at memory location 0x0000009855F4FA00. This happens when executing the write function for the Image class.

Currently looking into more about delegates.

**Comment by danielzx2** [comment, marked as answer]
Found the cause. 

Was running the program in "Debug" configuration, not "Release". Swapping it to "Release" fixed it and it works now.
